### ISSUE TITLE
Undefined Offset - Error

### ISSUE THREAD

**Issue by nicoschreiner** [firefly-iii/firefly-iii#1085]
I am running Firefly III version 4.6.11.1 and 4.7.0 (tested on demo page too - may you got an email?)

#### Description of my issue:
I run into an `Undefined offset`-error which leads to a warning-sign in the graphs `Expenses by category` and `Income by category`.

![image](https://user-images.githubusercontent.com/31184224/34468266-84645a52-ef05-11e7-884c-7712262e81ec.png)

#### Steps to reproduce 

1. Create a new `Category`.
2. Create a new `Withdrawal` and / or `Deposit` with this category.
3. Delete the `Category`.
4. Go to `Accounts` -> `Asset accounts` -> Click the account to which the withdrawal and deposit was added.
5. May you need to click on `Show all transactions without grouping them by date`.

#### Other important details (log files, system info):

```
Truncated from this point <----|[2018-01-01 13:59:55] production.ERROR: Exception is: {"class":"ErrorException","errorMessage":"Undefined offset: 4","time":"Mon, 01 Jan 2018 13:59:55 +0000","file":"\/srv\/firefly-iii\/www\/public_html\/firefly-iii\/app\/Http\/Controllers\/Chart\/AccountController.php","line":318,"code":0,"version":"4.6.11.1"}  
[2018-01-01 14:00:02] production.ERROR: Undefined offset: 4 {"userId":3,"email":"***","exception":"[object] (ErrorException(code: 0): Undefined offset: 4 at /srv/firefly-iii/www/public_html/firefly-iii/app/Http/Controllers/Chart/AccountController.php:318)"} 
[2018-01-01 14:00:10] production.ERROR: Exception is: {"class":"ErrorException","errorMessage":"Undefined offset: 4","time":"Mon, 01 Jan 2018 14:00:10 +0000","file":"\/srv\/firefly-iii\/www\/public_html\/firefly-iii\/app\/Http\/Controllers\/Chart\/AccountController.php","line":318,"code":0,"version":"4.6.11.1"}  
[2018-01-01 14:00:14] production.ERROR: Undefined offset: 4 {"userId":3,"email":"***","exception":"[object] (ErrorException(code: 0): Undefined offset: 4 at /srv/firefly-iii/www/public_html/firefly-iii/app/Http/Controllers/Chart/AccountController.php:318)"} 
[2018-01-01 14:01:11] production.ERROR: Exception is: {"class":"ErrorException","errorMessage":"Undefined offset: 4","time":"Mon, 01 Jan 2018 14:01:11 +0000","file":"\/srv\/firefly-iii\/www\/public_html\/firefly-iii\/app\/Http\/Controllers\/Chart\/AccountController.php","line":318,"code":0,"version":"4.6.11.1"}  
[2018-01-01 14:01:15] production.ERROR: Undefined offset: 4 {"userId":3,"email":"***","exception":"[object] (ErrorException(code: 0): Undefined offset: 4 at /srv/firefly-iii/www/public_html/firefly-iii/app/Http/Controllers/Chart/AccountController.php:318)"} 

```

**Comment by JC5** [comment]
Ah darn, I fixed a similar bug for budgets, but not for categories. A fix is on the way, and will go live in 4.7.0

**Comment by nicoschreiner** [comment]
Thanks! :D
